http/responseemitter.go

```diff
@@ -208,6 +229,22 @@ func (re *responseEmitter) doPreamble(value interface{}) {
 		mime   string
 	)
 
+	// If we have a request body, make sure we close the body
+	// if we want to write before completing reading.
+	// FIXME: https://github.com/ipfs/go-ipfs/issues/5168
+	// FIXME: https://github.com/golang/go/issues/15527
+	if re.bodyErrChan != nil {
+		select {
+		case <-re.bodyErrChan:
```

Comment:
I believe we can get transient errors (e.g., read timeouts). We probably do just want to check for an EOF and assume that other errors mean there may be more data. In all likelihood, any non-EOF error means the connection is in trouble anyways.